app/uk/gov/hmrc/economiccrimelevyregistration/controllers/AmendmentRequestedController.scala

```diff
@@ -44,7 +45,15 @@ class AmendmentRequestedController @Inject() (
       .get(request.session, request.internalId, SessionKeys.FirstContactEmailAddress)
       .map {
         case Some(firstContactEmailAddress) =>
-          Ok(view(firstContactEmailAddress, request.eclRegistrationReference))
+          val contactAddress =
+            Json.fromJson[EclAddress](Json.parse(request.session.get(SessionKeys.contactAddress).getOrElse(""))).asOpt
```

Comment:
can we use the session service instead - it returns a none when it doesn't find the session key

Reply:
Tried that but I had a problem with the `Future`. Also we already use the session for other things on this view.

Reply:
Made the change.